tests/lightgbm/test_lightgbm_autolog.py

```diff
@@ -312,3 +315,104 @@ def test_lgb_autolog_loads_model_from_artifact(bst_params, train_set):
     np.testing.assert_array_almost_equal(
         model.predict(train_set.data), loaded_model.predict(train_set.data)
     )
+
+
+@pytest.mark.large
+def test_lgb_autolog_gets_input_example(bst_params):
+    # we need to check the example input against the initial input given to train function.
+    # we can't use the train_set fixture for this as it defines free_raw_data=False but this
+    # feature should work even if it is True
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    dataset = lgb.Dataset(X, y, free_raw_data=True)
+
+    mlflow.lightgbm.autolog()
+    lgb.train(bst_params, dataset)
+    run = get_latest_run()
+
+    model_path = os.path.join(run.info.artifact_uri, "model")
+    model_conf = Model.load(os.path.join(model_path, "MLmodel"))
+
+    input_example = _read_example(model_conf, model_path)
+
+    assert input_example.equals(X[:5])
+
+    pyfunc_model = mlflow.pyfunc.load_model(os.path.join(run.info.artifact_uri, "model"))
+
+    # make sure reloading the input_example and predicting on it does not error
+    pyfunc_model.predict(input_example)
+
+
+@pytest.mark.large
+def test_lgb_autolog_infers_model_signature_correctly(bst_params):
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    dataset = lgb.Dataset(X, y, free_raw_data=True)
+
+    mlflow.lightgbm.autolog()
+    lgb.train(bst_params, dataset)
+    run = get_latest_run()
+    run_id = run.info.run_id
+    artifacts_dir = run.info.artifact_uri.replace("file://", "")
+    client = mlflow.tracking.MlflowClient()
+    artifacts = [x.path for x in client.list_artifacts(run_id, "model")]
+
+    ml_model_filename = "MLmodel"
+    assert str(os.path.join("model", ml_model_filename)) in artifacts
+    ml_model_path = os.path.join(artifacts_dir, "model", ml_model_filename)
+
+    data = None
+    with open(ml_model_path, "r") as f:
+        data = yaml.load(f, Loader=yaml.FullLoader)
+
+    assert data is not None
+    assert "signature" in data
+    signature = data["signature"]
+    assert signature is not None
+
+    assert "inputs" in signature
+    assert json.loads(signature["inputs"]) == [
+        {"name": "sepal length (cm)", "type": "double"},
+        {"name": "sepal width (cm)", "type": "double"},
+    ]
+
+    assert "outputs" in signature
+    assert json.loads(signature["outputs"]) == [
+        {"type": "double"},
+        {"type": "double"},
+        {"type": "double"},
+    ]
+
+
+@pytest.mark.large
+def test_lgb_autolog_continues_logging_even_if_signature_inference_fails():
+    # signature and input example inference should fail here since the dataset is given
+    #   as a file path
+    dataset = lgb.Dataset(os.path.join(os.path.dirname(__file__), "iris_training.csv"))
```

Comment:
Can we use a temporary csv file instead of `iris_training.csv`?

```suggestion
@pytest.mark.large
def test_lgb_autolog_continues_logging_even_if_signature_inference_fails(tmpdir):
    # signature and input example inference should fail here since the dataset is given
    #   as a file path
    tmp_csv = tmpdir.join("data.csv")
    tmp_csv.write("foo\n1.0")
    dataset = lgb.Dataset(tmp_csv.strpath)
```

Reply:
how `tmpdir` works:

https://docs.pytest.org/en/stable/tmpdir.html#the-tmpdir-fixture